### PR TITLE
Fix mk1 bt overflow

### PR DESCRIPTION
**Pull request by stieg**
Fixes overflow issues in this unit by

* Making BT task a higher priority
* Increasing BT queue tx/rx queue size

Also adds a little bit of debugging.

### CODE REVIEW

**Comment by brentpicasso** [comment]
From the description it should look like a few numbers adjusted but actually  includes a significant last minute refactoring that can arguably saved for a future release. Thoughts?

File: SAM7s_base/usart_at91/usart_device_at91.c [hunk 1]
```diff
@@ -141,9 +134,17 @@ static unsigned int createUartMode(unsigned int bits, unsigned int parity, unsig
...
     AT91F_PMC_EnablePeriphClock( AT91C_BASE_PMC,(1 << AT91C_ID_US0) | (1 << AT91C_ID_US1));
-    if (!initQueues()) return 0;
+
+    /* Init device 0 (Telemetry) */
+    if (!init_txrx_queue(0, 512, 512))
+            return 0;
     usart_device_init_0(8, 0, 1, TELEMETRY_BAUDRATE);
+
+    /* Init device 1 (GPS) */
+    if (!init_txrx_queue(1, 512, 128))
```
Comment (brentpicasso) [inline SAM7s_base/usart_at91/usart_device_at91.c]: Unrelated change with significant risk of GPS regression.  GPS Rx Buffer was reduced from 300 to 128; Tx Buffer increased from 300 512. MK1 treats GPS as a read only device; does not communicate with GPS at all.

**Comment by brentpicasso** [comment]
Rejecting this PR due to inclusion of unrelated changes, including at least one significant regression risk (GPS)

**Comment by stieg** [comment]
The re factor I did because it would have been an ugly hack to not do it.  If you look at the meat of the change most of it was eliminating duplicate code and consolidating code down.  So low risk there.  I will agree with you RE the GPS buffer settings.  I can set them back to 300/300 if you like, or do something like 512/300 (Rx/Tx) so that the chance of regression there is eliminated.  I don't feel like the code is high risk because if something was wrong it would be immediately obvious.

**Comment by stieg** [comment]
Fine.  I'll rebase this against 2.8.6.

**Comment by brentpicasso** [comment]
The refactoring work was sound in concept, definitely. But in principal, would be safest to put against 2.8.6.

Since MK1 does not communicate with GPS module *at all* it would be best to give it a generous Rx queue (say, 512) and a minimal Tx queue (say, 32).

**Comment by stieg** [comment]
> Since MK1 does not communicate with GPS module at all it would be best to give it a generous Rx queue (say, 512) and a minimal Tx queue (say, 32). 

That is exactly what I did.  And I thought it sends the GPS some data in the beginning to set NMEA parameters and data rate etc.
